Monika After Story/game/script-topics.rpy

```diff
@@ -15990,3 +15990,117 @@ label monika_stargazing:
     show monika 5kkbsa
     pause 2.0
     return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_player_eyesight",
+            category=["you"],
+            prompt="Wearing glasses",
+            conditional="mas_seenLabels(['monika_player_appearance'])"
+        )
+    )
+
+default persistent._mas_pm_wears_glasses = None
+default persistent._mas_pm_wears_lenses = None
+default persistent._mas_pm_poor_eyesight = None
+
+label monika_player_eyesight:
+    m 3lsb "Hey, [player], I was wondering about something lately."
+    m 1esa "Do you wear glasses, or maybe contact lenses?{nw}"
+    $ _history_list.pop()
+    menu:
+        m "Do you wear glasses, or maybe contact lenses?{fast}"
+
+        "I wear glasses.":
+            call monika_player_eyesight_awea(glasses=True)
+
+        "I wear contact lenses.":
+            call monika_player_eyesight_awea(glasses=False)
+
+        "I wear cosmetic glasses.":
+            call monika_player_eyesight_cg
+
+        "I wear glasses occasionally.":
+            call monika_player_eyesight_wgo
+
+        "Nope, I don't.":
+            call monika_player_eyesight_n
+
+    return "derandom"
+
+# Always Wears Eyesight Aids
+label monika_player_eyesight_awea(glasses=False):
+    if glasses:
+        $ persistent._mas_pm_wears_glasses = True
+    else:
+        $ persistent._mas_pm_wears_lenses = True
+    $ persistent._mas_pm_poor_eyesight = True
+    m 2ekc "Oh, I see. "
+    extend 2eka "I really hope you see better with them on."
+    $ aids = "glasses" if glasses else "contact lenses"
+    m 1eka "Please don't neglect your eyesight, [player], as it can still deteriorate even if you constantly wear your [aids]."
+    m 1esa "If you've been on your computer for a while, maybe you could consider taking a little break to let your eyes rest."
+    m 3hsa "It'll take you just a few minutes a day, but I believe it would really help."
+    m 1ekc "Although... if it gets really bad, go see a doctor, okay?"
+    m 1eka "I really want my sweetheart to see this beautiful world just as clear as I do~"
+    return
+
+# Cosmetic Glasses
+label monika_player_eyesight_cg:
+    $ persistent._mas_pm_wears_glasses = True
+    $ persistent._mas_pm_wears_lenses = False
+    $ persistent._mas_pm_poor_eyesight = False
+    m 5ekbsa "Aw, [player], I wish I really could see your face with them on."
+    m 5hubsa "I bet you look so smart, intelligent and confident~"
+    m 1sua "Maybe I could even get myself a pair and try them out, too, ehehe."
+    m 1tuu "What do you think, [mas_get_player_nickname()]? Would they suit me well, hmm?"
+    return
+
+# Wears Glasses Occasionally
+label monika_player_eyesight_wgo:
```

Comment:
Why don't you set `_mas_pm_wears_glasses` and `_mas_pm_wears_lenses ` here?

Reply:
I do, however, I believe contact lenses are not something you can just put on and off every now and then like glasses, thus this branch only sets `_mas_pm_wears_glasses` to `True`.

Reply:
> contact lenses are not something you can just put on and off every now and then

Pretty sure they are, but that's not the point. The player chose this option, and you should set the vars accordingly. In this case: glasses - `True`, lenses - `False`.